nixos/modules/services/web-servers/phpfpm/default.nix

```diff
@@ -48,19 +41,140 @@ let
     cat $phpPackage/etc/php.ini $nixDefaultsPath $phpOptionsPath > $out
   '';
 
+  poolOpts = { lib, name, ... }:
+    let
+      poolOpts = cfg.pools."${name}";
+    in
+    {
+      options = {
+        socket = mkOption {
+          type = types.str;
+          readOnly = true;
+          description = ''
+            Path to the unix socket file on which to accept FastCGI requests.
+            <note><para>This option is read-only and managed by NixOS.</note></para>
+          '';
+        };
+
+        listen = mkOption {
+          type = types.str;
```

Comment:
`types.path` can be used

Reply:
It also could be a tcp value so should be `str`.